core/src/accounts_cleanup_service.rs

```diff
@@ -0,0 +1,38 @@
+// Service to clean up dead slots in accounts_db
+//
+// This can be expensive since we have to walk the append vecs being cleaned up.
+
+use solana_ledger::bank_forks::BankForks;
+use std::sync::{
+    atomic::{AtomicBool, Ordering},
+    Arc, RwLock,
+};
+use std::thread::{self, sleep, Builder, JoinHandle};
+use std::time::Duration;
+
+pub struct AccountsCleanupService {
+    t_cleanup: JoinHandle<()>,
+}
+
+impl AccountsCleanupService {
+    pub fn new(bank_forks: Arc<RwLock<BankForks>>, exit: &Arc<AtomicBool>) -> Self {
+        info!("AccountsCleanupService active");
+        let exit = exit.clone();
+        let t_cleanup = Builder::new()
+            .name("solana-accounts-cleanup".to_string())
+            .spawn(move || loop {
+                if exit.load(Ordering::Relaxed) {
+                    break;
+                }
+                let bank = bank_forks.read().unwrap().working_bank();
```

Comment:
Hm, nice! This won't cause any leaks. You know, banks are scary... :p too easy to leak.

Reply:
Yea, should be fine, they are in an Arc, so it should dec ref when the bank goes out of scope here.